include/claim_strategy.hpp

```diff
@@ -0,0 +1,300 @@
+#ifndef GNURADIO_CLAIM_STRATEGY_HPP
+#define GNURADIO_CLAIM_STRATEGY_HPP
+
+#include <concepts>
+#include <cstdint>
+#include <memory>
+#include <span>
+#include <stdexcept>
+#include <vector>
+
+#ifndef GNURADIO_SEQUENCE_HPP
+#include <sequence.hpp>
+#endif
+#ifndef GNURADIO_WAIT_STRATEGY_HPP
+#include <wait_strategy.hpp>
+#endif
+
+namespace gr {
+
+struct NoCapacityException : public std::runtime_error {
+    NoCapacityException() : std::runtime_error("NoCapacityException"){};
+};
+
+// clang-format off
+
+template<typename T>
+inline constexpr bool isClaimStrategy = requires(T /*const*/ t, const std::vector<std::shared_ptr<Sequence>> &dependents, const int requiredCapacity,
```

Comment:
Why define both the type trait and the concept?

Reply:
fixed